compiler/src/dotty/tools/dotc/transform/SourcePositions.scala

```diff
@@ -0,0 +1,40 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.core.Constants.Constant
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Decorators._
+import dotty.tools.dotc.core.StdNames._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.transform.MegaPhase._
+
+import java.nio.file.Paths
+
+/** Replace references of dotty.source.Position.XYZ by their call site positions **/
+class SourcePositions extends MiniPhase {
+  import tpd._
+
+  def phaseName: String = "sourcePositions"
+
+  override def transformSelect(tree: Select)(implicit ctx: Context): Tree =
+    transformPosition(tree)
+
+  override def transformIdent(tree: Ident)(implicit ctx: Context): Tree =
+    transformPosition(tree)
+
+  private def transformPosition(tree: RefTree)(implicit ctx: Context): Tree = {
+    if (tree.symbol.owner ne defn.DottySourcePositionModuleRef.termSymbol.moduleClass) tree
+    else if (tree.symbol.name == nme.thisLine) Literal(Constant(tree.pos.line + 1)).asInstance(tree.tpe)
+    else if (tree.symbol.name == nme.thisSource) Literal(Constant(sourceFile)).asInstance(tree.tpe)
+    else tree
+  }
+
+  private def sourceFile(implicit ctx: Context): String = {
+    // TODO This could be performed one per unit to improve performance. Worth it?
+    // TODO Add a compiler flag to output absolute paths?
+    val currentPath = Paths.get(".").toAbsolutePath.getParent
```

Comment:
Is doing `Paths.get(".")` safe for all the possible filesystems that NIO supports? (I'm asking, I'm not sure myself). I think a compiler flag here would be a more elegant solution, and could be configured by default from the build tool.

Reply:
I’m not sure if this is safe I like to test with JimFS virtual filesystem as it only implements the NIO spec and no additional features.  The JVM implementation oddly implements a few things that are not part of NIO on all platforms.

Reply:
I will remove the `Paths.get(".")` and use a custom base path as suggested by @jvican